core/controllers/access_validators_test.py

```diff
@@ -416,3 +418,59 @@ def test_validation_returns_false_if_given_user_is_non_existent(
             ), expected_status_int=404
         )
         self.logout()
+
+
+class TopicEditorPageAccessValidationHandlerTests(test_utils.EmailTestBase):
+    """Checks the access to the topic editor page and its rendering."""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.topic_id = topic_fetchers.get_new_topic_id()
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+        self.skill_id_2 = skill_services.get_new_skill_id()
+        self.skill_id = skill_services.get_new_skill_id()
+
+    def test_access_topic_editor_page_without_logging_in(self) -> None:
+        self.get_json(
+            '%s/can_access_topic_editor/%s' %(
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+
+    def test_access_topic_editor_page_with_guest_user(self) -> None:
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' %(
+            ACCESS_VALIDATION_HANDLER_PREFIX,
+            self.topic_id), expected_status_int=401)
+        self.logout()
+
+        self.login(self.NEW_USER_EMAIL)
+        self.get_json(
+            '%s/can_access_topic_editor/%s' % (
+                ACCESS_VALIDATION_HANDLER_PREFIX, self.topic_id),
+            expected_status_int=401)
+        self.logout()
+
+    def test_access_topic_editor_page_with_curriculum_admin(
+        self
+    ) -> None:
+        self.login(self.CURRICULUM_ADMIN_EMAIL)
+        self.get_html_response(
```

Comment:
Why are we looking for HTML response here?

Reply:
Do we need to look at json response?
getting same error with json respons

Reply:
can you check wether moving this to get_json resolves the issue?

Reply:
just did in my local got this error again

```
=====================================================================
ERROR: test_access_topic_editor_page_with_curriculum_admin (core.controllers.access_validators_test.TopicEditorPageAccessValidationHandlerTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/harshkeshari/opensource/oppia/core/controllers/access_validators_test.py", line 463, in test_access_topic_editor_page_with_curriculum_admin
    self.get_json(
  File "/Users/harshkeshari/opensource/oppia/core/tests/test_utils.py", line 2885, in get_json
    json_response = self.testapp.get(
  File "/Users/harshkeshari/opensource/.direnv/python-3.8/lib/python3.8/site-packages/webtest/app.py", line 324, in get
    return self.do_request(req, status=status,
  File "/Users/harshkeshari/opensource/.direnv/python-3.8/lib/python3.8/site-packages/webtest/app.py", line 646, in do_request
    self._check_status(status, res)
  File "/Users/harshkeshari/opensource/.direnv/python-3.8/lib/python3.8/site-packages/webtest/app.py", line 680, in _check_status
    raise AppError(
webtest.app.AppError: Bad response: 404 Not Found (not 200)
)]}'
{"error": "Could not find the page http://localhost/access_validation_handler/can_access_topic_editor/7EMjW9QmKqzm.", "status_code": 404}

---------------------------------------------------------------------
FAILED    core.controllers.access_validators_test.TopicEditorPageAccessValidationHandlerTests: 1 errors, 0 failures

Ran 4 tests in 1 test class.
(1 ERRORS, 0 FAILURES)
Traceback (most recent call last):
  File "/Users/harshkeshari/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/harshkeshari/.pyenv/versions/3.8.15/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/Users/harshkeshari/opensource/oppia/scripts/run_backend_tests.py", line 683, in <module>
    main()
  File "/Users/harshkeshari/opensource/oppia/scripts/run_backend_tests.py", line 592, in main
    raise Exception(
Exception: 1 errors, 0 failures
```

Reply:
Hi @seanlip 
the URL is correct, I intentionally disturbed the test of `BlogAuthorProfilePageAccessValidationHandlerTests` just to check if my URL followed the same pattern as other tests.
![Screenshot 2024-02-20 at 9 36 22 AM](https://github.com/oppia/oppia/assets/108923011/28d51788-87b2-419d-b468-6f048e7b857d)

and got the error 
![Screenshot 2024-02-20 at 9 36 33 AM](https://github.com/oppia/oppia/assets/108923011/953d356c-4097-4f34-9d78-15eee72e10e7)